modules/ozoneBidAdapter.js

```diff
@@ -263,46 +313,195 @@ export const spec = {
+      arrQueryString.push('cb=' + Date.now());
+
+      var strQueryString = arrQueryString.join('&');
+      if (strQueryString.length > 0) {
+        strQueryString = '?' + strQueryString;
+      }
+      utils.logInfo('OZONE: getUserSyncs going to return cookie sync url : ' + OZONECOOKIESYNC + strQueryString);
       return [{
         type: 'iframe',
-        url: OZONECOOKIESYNC
+        url: OZONECOOKIESYNC + strQueryString
       }];
     }
+  },
+
+  /**
+   *  Look for pubcid & all the other IDs according to http://prebid.org/dev-docs/modules/userId.html
+   *  @return map
+   */
+  findAllUserIds(bidRequest) {
+    var ret = {};
+    let searchKeysSingle = ['pubcid', 'tdid', 'id5id', 'parrableid', 'idl_env', 'digitrustid', 'criteortus'];
+    utils.logInfo('OZONE: debug iterating keys');
+    utils.logInfo('OZONE: debug bidRequest=', bidRequest);
+    if (bidRequest.hasOwnProperty('userId')) {
+      utils.logInfo('OZONE: debug Looking inside userId element');
+      for (let arrayId in searchKeysSingle) {
+        let key = searchKeysSingle[arrayId];
+        utils.logInfo('OZONE: debug key=', key);
+        if (bidRequest.userId.hasOwnProperty(key)) {
+          utils.logInfo('OZONE: debug found value : ', key);
+          ret[key] = bidRequest.userId[key];
+        }
+      }
+      var lipbid = utils.deepAccess(bidRequest.userId, 'lipb.lipbid');
+      if (lipbid) {
+        ret['lipb'] = {'lipbid': lipbid};
+      }
+    }
+    if (!ret.hasOwnProperty('pubcid')) {
+      var pubcid = utils.deepAccess(bidRequest, 'crumbs.pubcid');
+      if (pubcid) {
+        ret['pubcid'] = pubcid; // if built with old pubCommonId module
+      }
+    }
+    utils.logInfo('OZONE: debug going to return: ', ret);
+    return ret;
+  },
+
+  /**
+   * Convenient method to get the value we need for the placementId - either from the bidRequest, or from the GET
+   * parameter introduced in 2.2.0 : ozstoredrequest
+   * IF the GET parameter exists then it must validate for placementId correctly
+   * @param bidRequest
+   * @return string
+   */
+  getPlacementId(bidRequest) {
+    let arr = this.getGetParametersAsObject();
+    this.placementIdWasOverridenByGetParam.val = 0;
+    if (arr.hasOwnProperty('ozstoredrequest')) {
+      if (this.isValidPlacementId(arr.ozstoredrequest)) {
+        let override = arr.ozstoredrequest;
+        utils.logInfo('OZONE: using GET ozstoredrequest ' + override + ' to replace placementId');
+        this.placementIdWasOverridenByGetParam.val = 1;
+        return override;
+      } else {
+        utils.logError('OZONE: GET ozstoredrequest FAILED VALIDATION - will not use it');
+      }
+    }
+    return (bidRequest.params.placementId).toString();
+  },
+  /**
+   * Produces external userid object
+   */
+  addExternalUserId(eids, value, source, atype) {
+    if (utils.isStr(value)) {
+      eids.push({
+        source,
+        uids: [{
+          id: value,
+          atype
```

Comment:
@afsheenb I was going through this code for another PR I'm working on (to convert our ID5 ID from a string to an object, which would require a small change to your adapter...I'll tag you in it when it's ready).

In this code here, I don't think you're generating a valid json object. the `source` and `atype` values passed in to this function from below are just strings/numbers, so you're missing the keys for those items. I think the code here should look like this instead:

```javascript
eids.push({
  source: source,
  uids: [{
    id: value,
    atype: atype
  }]
});
```